darts/models/exponential_smoothing.py

```diff
@@ -75,11 +76,15 @@ def fit(self, series: TimeSeries):
                                            trend=self.trend.value,
                                            damped_trend=self.damped,
                                            seasonal=self.seasonal.value,
-                                           seasonal_periods=self.seasonal_periods)
-
+                                           seasonal_periods= None if self.infer_seasonal_periods else self.seasonal_periods,
+                                           freq=series.freq,
+                                           dates=series.time_index)
```

Comment:
Will this work if the series is integer-indexed (and its index is a `pandas.Int64Index`)? From the statsmodels doc it looks like this expects rather a DateTimeIndex. I think we should probably condition this inference to cases where the series are datetime-indexed only, and raise an error if the seasonality is not specified and the series is integer-indexed.